cirq-core/cirq/ops/common_gates.py

```diff
@@ -237,10 +237,12 @@ def _circuit_diagram_info_(
 
     def _qasm_(self, args: 'cirq.QasmArgs', qubits: Tuple['cirq.Qid', ...]) -> Optional[str]:
         args.validate_version('2.0')
-        if self._exponent == 1 and self._global_shift != -0.5:
+        if self._exponent % 2 == 1 and self._global_shift != -0.5:
             return args.format('x {0};\n', qubits[0])
-        elif self._exponent == 0.5:
+        elif self._exponent % 2 == 0.5:
             return args.format('sx {0};\n', qubits[0])
+        elif self._exponent % 2 == 1.5:
+            return args.format('sxdg {0};\n', qubits[0])
```

Comment:
There is a general question here: to what extent should `_qasm_` be simplifying the circuit using known gate properties like X**2=I? One one hand, we could argue that we should preserve programmer's intention in writing something like rx(2.5pi) on the grounds that perhaps they know some low-level hardware details (e.g. gate timing or noise) and simplification would deny them a useful control knob. On the other hand, we could argue for mathematical simplicity and predictability for example to enable the consumer of the QASM to make some reasonable assumptions about the range of parameters such as rotation angles (e.g. they may expect RX rotation angle to be in say [0, 2pi)).

The mathematician in me prefers the latter, but the hacker wants the former. I could live with either. However, the current code bothers me a little bit since by failing to apply simplification for the case of even integer exponents it does not take a uniform stance on the above question.

Suggestions:
1. To make the mathematician happy: handle `self._exponent % 2 == 0`, e.g. by returning empty string (if this is a valid behavior for `_qasm_`) or explicit identity gate or at least "rx(0)" for all even integer exponents.
2. To make the hacker happy: remove `% 2`, i.e. preserve the old treatment of exponents.

My preference is for 2, but I leave the choice up to you.

Reply:
A variant of option 1 also changes

```python
return args.format('rx({0:half_turns}) {1};\n', self._exponent, qubits[0])
```

to 

```python
return args.format('rx({0:half_turns}) {1};\n', self._exponent % 2, qubits[0])
```

Reply:
Chose 2.  I also chose to make it `X**(-0.5)` as that feels like the most likely use case.  

It feels to me like this is one of those places where we should have had a design where the gate stuff sits on one side and the serialization / instruction set stuff sits on the other side, then there might be default way to use gate stuff to do instruction stuff, but this could be mediated by something that can make other decisions.

Reply:
+1 to `X**(-0.5)`

Reply:
Yeah, I agree it'd be great to have a design clearly stating such things, e.g. adopting or rejecting a principle like

> serialization code must be faithful at the parameter level, not merely at the gate algebra level.

As it is, I'm not even sure cirq's behavior is consistent across various gates.